plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveConnectorTest.java

```diff
@@ -1618,6 +1620,40 @@ public void testCreateTableWithUnsupportedType()
         assertQueryFails("CREATE TABLE test_create_table_with_unsupported_type AS SELECT TIME '00:00:00' x", "\\QUnsupported Hive type: time(0)\\E");
     }
 
+    @Test
+    public void testTargetMaxFileSize()
+    {
+        @Language("SQL") String createTableSql = "CREATE TABLE test_max_file_size AS SELECT * FROM tpch.sf1.lineitem LIMIT 1000000";
+        @Language("SQL") String selectFileInfo = "SELECT distinct \"$path\", \"$file_size\" FROM test_max_file_size";
+
+        // verify the default behavior is one file per node
+        Session session = Session.builder(getSession())
+                .setSystemProperty("task_writer_count", "1")
+                .build();
+        assertUpdate(session, createTableSql, 1000000);
+        assertThat(computeActual(selectFileInfo).getRowCount()).isEqualTo(3);
+        assertUpdate("DROP TABLE test_max_file_size");
+
+        // Write table with small limit and verify we get multiple files per node near the expected size
+        // Writer writes chunks of rows that are about 40k
+        DataSize maxSize = DataSize.of(40, Unit.KILOBYTE);
+        session = Session.builder(getSession())
+                .setSystemProperty("task_writer_count", "1")
+                .setCatalogSessionProperty("hive", "target_max_file_size", maxSize.toString())
+                .setCatalogSessionProperty("hive", "hive_storage_format", "TEXTFILE")
```

Comment:
please add a comment why `TEXTFILE` is important here (or drop if it is not). Also set same `hive_storage_format` here and in session used for sanity check above.